File: competitive-events-api/src/main/java/es/jocasolo/competitiveeventsapi/dto/event/EventPostDTO.java
package es.jocasolo.competitiveeventsapi.dto.event;

import java.io.Serializable;
import java.util.Date;

import es.jocasolo.competitiveeventsapi.dto.DTO;
import es.jocasolo.competitiveeventsapi.enums.event.EventInscriptionType;
import es.jocasolo.competitiveeventsapi.enums.event.EventType;
import es.jocasolo.competitiveeventsapi.enums.event.EventVisibilityType;
import es.jocasolo.competitiveeventsapi.enums.score.ScoreSortType;
import es.jocasolo.competitiveeventsapi.enums.score.ScoreValueType;
import lombok.Getter;
import lombok.Setter;

public class EventPostDTO extends DTO implements Serializable {

	private static final long serialVersionUID = 1L;

	@Getter
	@Setter
	private String title;

	@Getter
	@Setter
	private String subtitle;

	@Getter
	@Setter
	private String description;

	@Getter
	@Setter
	private EventType type;

	@Getter
	@Setter
	private EventInscriptionType inscription;

	@Getter
	@Setter
	private EventVisibilityType visibility;

	@Getter
	@Setter
	private Boolean approvalNeeded;

	@Getter
	@Setter
	private Date initDate;

	@Getter
	@Setter
	private Date endDate;

	@Getter
	@Setter
	private Integer maxPlaces;

	@Getter
	@Setter
	private ScoreValueType scoreType;
	
	@Getter
	@Setter
	private ScoreSortType sortScore;

	@Override
	public String toString() {
		return "EventPostDTO [title=" + title + "]";
	}

}

File: competitive-events-app/app/src/main/java/es/jocasolo/competitiveeventsapp/dto/event/EventPostDTO.kt
package es.jocasolo.competitiveeventsapp.dto.event

import com.google.gson.annotations.SerializedName
import es.jocasolo.competitiveeventsapp.enums.event.EventInscriptionType
import es.jocasolo.competitiveeventsapp.enums.event.EventType
import es.jocasolo.competitiveeventsapp.enums.event.EventVisibilityType
import es.jocasolo.competitiveeventsapp.enums.score.ScoreSortType
import es.jocasolo.competitiveeventsapp.enums.score.ScoreValueType

class EventPostDTO(
        @SerializedName("title") var title: String,
        @SerializedName("scoreType") var scoreType: ScoreValueType? = null,
        @SerializedName("sortScore") var sortScore: ScoreSortType? = null,
        @SerializedName("type") var type: EventType? = null,
        @SerializedName("inscription") var inscription: EventInscriptionType? = null,
        @SerializedName("visibility") var visibility: EventVisibilityType? = null,
        @SerializedName("approvalNeeded") var approvalNeeded: Boolean? = null,
        @SerializedName("subtitle") var subtitle: String? = null,
        @SerializedName("description") var description: String? = null,
        @SerializedName("initDate") var initDate: String? = null,
        @SerializedName("endDate") var endDate: String? = null,
        @SerializedName("maxPlaces") var maxPlaces: Int? = null
) {

}